pkg/controller/daemon/daemoncontroller.go

```diff
@@ -1016,6 +1016,30 @@ func (dsc *DaemonSetsController) syncDaemonSet(key string) error {
 	return dsc.updateDaemonSetStatus(ds)
 }
 
+// we try to partition predicates into two partitions: intentional on the part of the operator and not.
+// There is no need to fire event for intentional errors.
```

Comment:
The above comment is less relevant to this function but more relevant to the caller of this function. Update it to:
```go
// hasIntentionalPredicatesReasons checks if any of the given predicate failure reasons 
// is intentional. 
```